jf_agent/agent_logging.py

```diff
@@ -163,17 +191,28 @@ def handle(self, record: Union[logging.LogRecord, int]) -> None:
 
         if record != self._sentinel:
             msg = _standardize_log_msg(self.format(record))
-            self.messages_to_send.append(
-                {
-                    'message': msg,
-                    'timestamp': int(
+
+            # This logic is required to handle the different way that the timestamp attribute
+            # is set by structlog loggers
+            try:
+                asctime = (
+                    int(
                         datetime.strptime(
                             record.asctime + '000', '%Y-%m-%d %H:%M:%S,%f'
                         ).timestamp()
-                        * 1000
-                    ),
-                    'initiated_at': self.initiated_at,
-                }
+                    )
+                    * 1000
+                )
+            except AttributeError:
+                created_time = datetime.fromtimestamp(record.created)
+                asctime = created_time.strftime('%Y-%m-%d %H:%M:%S,%f')[:-3]
+
+            timestamp = int(
+                datetime.strptime(asctime + '000', '%Y-%m-%d %H:%M:%S,%f').timestamp() * 1000
```

Comment:
Ditto to what Jack said above, I'm also a little confused why were converting this to a string and then from a string back to a datetime?

I think you could go off of purely the `record.created` time, right? It comes back in already a timestamp formate (although we may still have to multiply it by 1000, I'm not sure about that)

https://docs.python.org/3/library/logging.html

Reply:
Agreed, using `.created` is much simpler, I've updated the code